packages/koa-metrics/src/test/Metrics.test.ts

```diff
@@ -0,0 +1,165 @@
+import withEnv from '@shopify/with-env';
+import {StatsD} from 'hot-shots';
+import Metrics from '../Metrics';
+
+jest.mock('hot-shots');
+const StatsDMock = StatsD as jest.Mock<StatsD>;
+
+describe('Metrics', () => {
+  beforeEach(() => {
+    StatsDMock.mockClear();
+  });
+
+  const defaultOptions = {
+    host: 'localhost',
+    port: 1234,
+    prefix: 'MyModule',
+    globalTags: {
+      aaa: 'A',
+    },
+  };
+
+  it('passes the host, port, prefix, and global tags to the statsd client', () => {
+    const metrics = new Metrics(defaultOptions);
+
+    expect(StatsDMock).toHaveBeenCalledTimes(1);
+    expect(StatsDMock).toHaveBeenCalledWith(defaultOptions);
+  });
+
+  describe('timing', () => {
+    it('passes timing metrics to the statsd client', () => {
+      const metrics = new Metrics(defaultOptions);
+      metrics.timing('foo', 123, {tag: 'value'});
+
+      const stats = StatsDMock.mock.instances[0];
+      const timingFn = stats.timing;
+      expect(timingFn).toHaveBeenCalledTimes(1);
+      expect(timingFn).toHaveBeenCalledWith('foo', 123, {tag: 'value'});
+    });
+
+    it('logs timing metrics to the logger in development', () => {
+      withEnv('development', () => {
+        const logger = jest.fn();
+        const metrics = new Metrics(defaultOptions, logger);
+        metrics.timing('foo', 123);
+
+        expect(logger).toHaveBeenCalledTimes(1);
+        expect(logger).toHaveBeenCalledWith('timing foo:123');
+      });
+    });
+
+    it('logs tags with timing metrics to the logger in development', () => {
+      withEnv('development', () => {
+        const logger = jest.fn();
+        const metrics = new Metrics(defaultOptions, logger);
+        metrics.timing('foo', 123, {tag: 'value'});
+
+        expect(logger).toHaveBeenCalledTimes(1);
+        expect(logger).toHaveBeenCalledWith('timing foo:123 #tag:value');
```

Comment:
should we be including the global tags in these logs?

Reply:
I don't *think* so